src/charm.py

```diff
@@ -240,23 +275,23 @@ def _create_discourse_environment_settings(self):
 
         saml_config = self._get_saml_config()
         for key in saml_config:
-            pod_config[key] = saml_config[key]
+            pod_config.update({key: saml_config[key]})
 
         if self.config.get("s3_enabled"):
             pod_config.update(self._get_s3_env())
 
         # We only get valid throttle levels here, otherwise it would be caught
-        # by `check_for_config_problems`, so we can be sure this won"t raise a
-        # KeyError.
-        for key in THROTTLE_LEVELS[self.config["throttle_level"]]:
-            pod_config[key] = THROTTLE_LEVELS[self.config["throttle_level"]][key]
+        # by `check_for_config_problems`.
+        if THROTTLE_LEVELS.get(self.config["throttle_level"]):
+            pod_config.update(THROTTLE_LEVELS.get(self.config["throttle_level"]))  # type: ignore
```

Comment:
Similar question here, what is the type issue?

Reply:
Will add some comments